packages/loader/src/index.ts

```
@@ -28,17 +28,48 @@ export const createChainedLoader: (loaders: Loader[]) => Loader = loaders => {
   };
 };
 
+function flattenStackTrace(err: Error | RendrError, stack: string[] = []) {
+  if (err instanceof RendrError && err.previousError) {
+    flattenStackTrace(err.previousError, stack);
+  }
+
+  stack.push(err.stack);
+
+  return stack;
+}
+
 // Simple version of a loader that catches errors during the execution of
 // following loaders
-// Feel free to use your own if you want to manage other error codes
+// Behaviour is as follow:
+// 1. If headers were sent, just ends the response
+// 2. The resultPage takes the same status code as the response if different from 200
+// 3. The resultPage receives the message of the error in the key message of its settings
+// 4. The context receives the error trace in the key stackTrace of its settings
+// 5. In non production mode, resultPage also receives the error trace
 export const errorBoundaryLoader: Loader = async (ctx, page, next) => {
   let resultPage: Page | void;
   try {
     resultPage = await next();
   } catch (err) {
+    if (ctx.res.headersSent) {
+      ctx.res.end();
+      return;
+    }
+
     resultPage = createPage();
     resultPage.statusCode = err instanceof NotFoundError ? 404 : 500;
+
+    if (ctx.res.statusCode >= 300) {
+      resultPage.statusCode = ctx.res.statusCode;
+    }
+
+    const fullStack = flattenStackTrace(err);
+    ctx.settings.stackTrace = fullStack;
     resultPage.settings.message = err.message;
+
+    if (ctx.isServerSide && process.env.NODE_ENV !== "production") {
+      resultPage.settings.stackTrace = fullStack;
```

We need to call a callback here, we cannot have silent error with no log.